Time::Weekday to Variant conversion problem on macOS build

### Godot version

917fd65748957304c987414c63d54ef4f6972394

### System information

macOS 10.14.6

### Issue description

Apple clang version 11.0.0 (clang-1100.0.33.17) (latest for 10.14.6)

Build fails on `Time::Weekday` conversion to `Variant`:
```
In file included from tests/test_main.cpp:70:
./tests/core/test_time.h:94:26: error: conversion from 'Time::Weekday' to 'const Variant' is ambiguous
        datetime[WEEKDAY_KEY] = Time::Weekday::WEEKDAY_SUNDAY;
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./core/variant/variant.h:385:2: note: candidate constructor
        Variant(bool p_bool);
        ^
./core/variant/variant.h:386:2: note: candidate constructor
        Variant(signed int p_int); // real one
        ^
./core/variant/variant.h:387:2: note: candidate constructor
        Variant(unsigned int p_int);
        ^
./core/variant/variant.h:392:2: note: candidate constructor
        Variant(signed short p_short); // real one
        ^
./core/variant/variant.h:393:2: note: candidate constructor
        Variant(unsigned short p_short);
        ^
./core/variant/variant.h:394:2: note: candidate constructor
        Variant(signed char p_char); // real one
        ^
./core/variant/variant.h:395:2: note: candidate constructor
        Variant(unsigned char p_char);
        ^
./core/variant/variant.h:396:2: note: candidate constructor
        Variant(int64_t p_int); // real one
        ^
./core/variant/variant.h:397:2: note: candidate constructor
        Variant(uint64_t p_int);
        ^
./core/variant/variant.h:398:2: note: candidate constructor
        Variant(float p_float);
        ^
./core/variant/variant.h:399:2: note: candidate constructor
        Variant(double p_double);
        ^
```

### Steps to reproduce

Using the following build arguments:

```
platform=osx arch=x86_64 target=debug dev=yes --jobs=$(sysctl -n hw.logicalcpu) 
```

### Minimal reproduction project

_No response_

--- a/tests/core/test_time.h
+++ b/tests/core/test_time.h
@@ -91,7 +91,7 @@ TEST_CASE("[Time] Datetime dictionary conversion methods") {
 	datetime[YEAR_KEY] = 2014;
 	datetime[MONTH_KEY] = 2;
 	datetime[DAY_KEY] = 9;
-	datetime[WEEKDAY_KEY] = Weekday::WEEKDAY_SUNDAY;
+	datetime[WEEKDAY_KEY] = (int64_t)Weekday::WEEKDAY_SUNDAY;
 	datetime[HOUR_KEY] = 22;
 	datetime[MINUTE_KEY] = 10;
 	datetime[SECOND_KEY] = 30;
@@ -100,7 +100,7 @@ TEST_CASE("[Time] Datetime dictionary conversion methods") {
 	date_only[YEAR_KEY] = 2014;
 	date_only[MONTH_KEY] = 2;
 	date_only[DAY_KEY] = 9;
-	date_only[WEEKDAY_KEY] = Weekday::WEEKDAY_SUNDAY;
+	date_only[WEEKDAY_KEY] = (int64_t)Weekday::WEEKDAY_SUNDAY;
 
 	Dictionary time_only;
 	time_only[HOUR_KEY] = 22;

